### PR TITLE
Allow clients to add additional information

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* Allowing clients to add additional information about their current situation, _in their own words_ has proven valuable for food assistance applications.

Changes
=======
* Add a step in the Preferences section at the end, just before signatures.

Minor
-----
* Remove warning for hash key defined twice.

### CODE REVIEW

**Comment by adarsh** [comment]
[As an applicant, I want to be able to share my story and situation in my own words.](https://trello.com/c/yClyAYp2/134-as-an-applicant-i-want-to-be-able-to-share-my-story-and-situation-in-my-own-words)

File: app/steps/step_navigation.rb [hunk 1]
```diff
@@ -35,6 +35,7 @@ class StepNavigation
     ],
     "Preferences" => [
       PreferencesInterviewController,
+      PreferencesAnythingElseController,
```
Comment (jessieay) [inline app/steps/step_navigation.rb]: do we think this controller should be considered part of the preferences section? Feels kinda separate to me

File: app/views/preferences_anything_else/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<% content_for :header_title, "What else should we know?" %>
```
Comment (jessieay) [inline app/views/preferences_anything_else/edit.html.erb]: Not sure if Alan suggested nav text. Usually we've been using the same nav text for each page in a section (this is not 100% consistent but it is the case probably 80% of the time)

File: app/views/preferences_anything_else/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+      <div class="step-section-header__subhead">
+        Feel free to give other details you think might be helpful for us to know. Otherwise, you can skip this question.
+      </div>
+    </div>
+  </header>
+
+  <div class="form-card__content">
+    <%= form_for @step, as: :step, builder: MbFormBuilder, url: current_path, method: :put do |f| %>
+      <%= f.mb_textarea :anything_else, "Anything else?", options: { placeholder: "Write here" } %>
```
Comment (jessieay) [inline app/views/preferences_anything_else/edit.html.erb]: I think there is an ongoing preference to not use placeholder text  https://trello.com/c/CstrMJ8l/85-design-as-a-client-i-want-the-most-usable-form-fields-possible-which-means-no-placeholder-text

File: app/views/preferences_anything_else/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<% content_for :header_title, "What else should we know?" %>
+
+<div class="form-card">
+  <header class="form-card__header">
+    <div class="step-section-header">
+      Is there anything else you'd like us to know about your situation?
+
+      <div class="step-section-header__subhead">
+        Feel free to give other details you think might be helpful for us to know. Otherwise, you can skip this question.
```
Comment (jessieay) [inline app/views/preferences_anything_else/edit.html.erb]: break at 80 chars?

File: db/migrate/20170824170659_add_anything_else_to_snap_application.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddAnythingElseToSnapApplication < ActiveRecord::Migration[5.1]
+  def change
+    add_column :snap_applications, :anything_else, :text
```
Comment (jessieay) [inline db/migrate/20170824170659_add_anything_else_to_snap_application.rb]: on GCF they call this field `additional_information` -- thoughts on that name?

File: spec/features/snap_app_spec.rb [hunk 6]
```diff
@@ -251,6 +256,11 @@
       click_on "Continue"
     end
 
+    on_page "What else should we know?" do
```
Comment (jessieay) [inline spec/features/snap_app_spec.rb]: maybe on this one we do not fill in to test at feature level that we do not need data in this field?

File: spec/models/snap_application_attributes_spec.rb [hunk 7]
```diff
@@ -53,7 +53,6 @@
         financial_accounts_mutual_funds_or_stocks: false,
         financial_accounts_four_oh_one_k_or_iras: true,
         financial_accounts_trusts: false,
-        first_member_with_disability_name: nil,
```
Comment (jessieay) [inline spec/models/snap_application_attributes_spec.rb]: removed on purpose?

File: app/steps/step_navigation.rb (hunk 1 above)
Comment (adarsh) [reply app/steps/step_navigation.rb]: Agreed. Renamed `Preferences` to `General` in a1b092f

File: app/views/preferences_anything_else/edit.html.erb (hunk 2 above)
Comment (adarsh) [reply app/views/preferences_anything_else/edit.html.erb]: Good idea! 9dc56f6

File: db/migrate/20170824170659_add_anything_else_to_snap_application.rb (hunk 5 above)
Comment (adarsh) [reply db/migrate/20170824170659_add_anything_else_to_snap_application.rb]: My thoughts are that it's a better name :) 2f30308

File: spec/models/snap_application_attributes_spec.rb (hunk 7 above)
Comment (adarsh) [reply spec/models/snap_application_attributes_spec.rb]: Yes, this was a duplicate and was throwing a Ruby-level warning for the duplicated key. It looks like someone else fixed this after I rebased!

**Pull request merged**